kimchi/src/circuits/polynomials/foreign_mul/common.rs

```diff
@@ -0,0 +1,24 @@
+//!
+//! Common constraints and related helpers
+//!
+
+use ark_ff::{FftField, One};
+
+use crate::circuits::expr::E;
+
+pub(crate) fn two<F: FftField>() -> E<F> {
+    2u64.into()
+}
+
+pub(crate) fn three<F: FftField>() -> E<F> {
+    3u64.into()
+}
+
+// Crumb constraint for 2-bit sublimb
+pub(crate) fn sublimb_crumb_constraint<F: FftField>(sublimb: &E<F>) -> E<F> {
```

Comment:
I was thinking of moving this as an Expr helper in a subsequent PR.  AFAIK, it only makes sense to use degree-n constraints up to n=4, so we could have a helper for these to avoid re-implementing all the time.  wdyt?